internal/serialization/class_definition_writer.go

```diff
@@ -68,26 +68,24 @@ func (cdw *ClassDefinitionWriter) WriteString(fieldName string, value string) {
 	cdw.classDefinitionBuilder.AddUTFField(fieldName)
 }
 
-func (cdw *ClassDefinitionWriter) WritePortable(fieldName string, portable serialization.Portable) error {
+func (cdw *ClassDefinitionWriter) WritePortable(fieldName string, portable serialization.Portable) {
 	if portable == nil {
-		return hzerror.NewHazelcastSerializationError("cannot write nil portable without explicitly registering class definition", nil)
+		panic(hzerror.NewHazelcastSerializationError("cannot write nil portable without explicitly registering class definition", nil))
```

Comment:
This function is about writing a portable value, so it's OK to panic if none is passed. The users still get an error if they pass a `nil` `portable` here, since the panic is caught in `ToData`. Other than that, my comments about our discussion on panics apply.

Reply:
Since it boils down to the same discussion as in https://github.com/hazelcast/hazelcast-go-client/pull/555, consider this comment to be a nit.